cli/coredump/coredump.go

```diff
@@ -0,0 +1,177 @@
+package coredump
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"embed"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/apex/log"
+	"github.com/tarantool/tt/cli/util"
+)
+
+//go:embed scripts/*
+var corescripts embed.FS
+
+// MakeTarGzReader makes reader for tar.gz file.
+func MakeTarGzReader(archive *os.File) (*tar.Reader, error) {
+	uncompressedStream, err := gzip.NewReader(archive)
+	if err != nil {
+		return nil, err
+	}
+
+	tarReader := tar.NewReader(uncompressedStream)
+	return tarReader, err
+}
+
+// ExtractTarGz extracts tar.gz archive.
+func ExtractTarGz(tarName string) error {
+	path, err := filepath.Abs(tarName)
+	if err != nil {
+		return err
+	}
+	archive, err := os.Open(path)
+	if err != nil {
+		return err
+	}
+	tarReader, err := MakeTarGzReader(archive)
+	if err != nil {
+		return err
+	}
+	for {
+		header, err := tarReader.Next()
+
+		if err == io.EOF {
+			break
+		} else if err != nil {
+			return err
+		}
+
+		switch header.Typeflag {
+		case tar.TypeDir:
+			continue
+		case tar.TypeReg:
+			var pos int
+			// Some archives have strange order of objects,
+			// so we check that all folders exist before
+			// creating a file.
+			pos = strings.LastIndex(header.Name, "/")
+			if pos == -1 {
+				pos = 0
+			}
+			if _, err := os.Stat(header.Name[0:pos]); os.IsNotExist(err) {
+				// 0755:
+				//    user:   read/write/execute
+				//    group:  read/execute
+				//    others: read/execute
+				os.MkdirAll(header.Name[0:pos], 0755)
+			}
+			outFile, err := os.Create(header.Name)
+			if err != nil {
+				return err
+			}
+			if _, err := io.Copy(outFile, tarReader); err != nil {
+				return err
+			}
+			outFile.Close()
+
+		default:
+			return fmt.Errorf("uknown type: %b in %s", header.Typeflag, header.Name)
+		}
+
+	}
+	archive.Close()
+	return nil
+}
+
+// Pack packs coredump into a tar.gz archive.
+func Pack(coreName string) error {
+	corePath, err := filepath.Abs(coreName)
+	if _, err := os.Stat(corePath); err != nil {
+		return fmt.Errorf("There was some problem packing archive. "+
+			"Error: '%v'.", err)
+	}
+	content, err := util.ReadEmbedFile(corescripts, "scripts/tarabart.sh")
+	if err != nil {
+		return err
+	}
+	cmd := exec.Command("sh", "-s")
```

Comment:
Maybe `bash` is better? As I understand it, `sh` may be a symlink to different shells and not all of them will work well with our scripts.

Reply:
Fixed